### PR TITLE
Skip assets if metadata is unchanged

### LINKED ISSUE #14: backups2datalad.py: skip update entirely if no changes according to stored metadata listing file
to make it possible to avoid inefficient file-by-file checks if we know that no file has changed from previous time

but there should be an option `--force=check` (later we might add other force)  which would cause full check

edit: talking about `.dandi/assets.json` which should not change if nothing changed on server side

### PR DESCRIPTION
**Pull request by jwodder**
Closes #14.

### CODE REVIEW

File: tools/backups2datalad.py [hunk 1]
```diff
@@ -214,13 +227,28 @@ def get_annex_hash(filepath):
...
                 dest = dsdir / a["path"].lstrip("/")
                 deststr = str(dest.relative_to(dsdir))
                 local_assets.discard(dest)
                 am = deepcopy(a)
                 am["modified"] = str(am["modified"])
                 asset_metadata.append(am)
+                if (
+                    self.force is None or "check" not in self.force
+                ) and am == saved_metadata.get(am["path"]):
+                    log.info(
```
Comment (yarikoptic) [inline tools/backups2datalad.py:247]: I will lower this one to `.debug` -- in a typical run it just fills up the screen, hard to monitor for "progress"

Comment (jwodder) [reply tools/backups2datalad.py]: Done.

**Comment by yarikoptic** [comment]
Takes about 8 minutes now for a full sweep, much better, should suffice for now

**Comment by yarikoptic** [comment]
note: I have made cron job to run each hour but with flock so we skip if prior run has not finished yet (e.g. fetching/pushing takes too long). so it looks like `flock -E 0 -e -n /tmp/backup2datalad-cron.lck /mnt/backup/dandi/dandisets/tools/backups2datalad-cron
`

**Pull request merged**